src/libsyntax/parse/parser.rs

```diff
@@ -2663,7 +2665,7 @@ impl<'a> Parser<'a> {
     }
 
     pub fn check_unknown_macro_variable(&mut self) {
-        if self.quote_depth == 0 {
+        if self.quote_depth == 0 && !self.parsing_token_tree {
```

Comment:
I believe `quote_depth` and `parsing_token_tree` have the exact same intended function.